PBM-1576-Release notes 2.10.0 (#287)
percona/pbm-docs: docs/release-notes/2.10.0.md

# Percona Backup for MongoDB 2.10.0 ({{date.2_10_0}})

[Installation](../installation.md){.md-button}

Percona Backup for MongoDB is a distributed, low-impact solution for creating consistent backups of MongoDB sharded clusters and replica sets, and for restoring those backups to a specific point in time.

## Release Highlights

This release provides the following features and improvements:

### Fallback strategy for physical restores to ensure cluster operation

> You can now configure a fallback directory for physical restores in PBM. At the restore start, PBM copies the `dbPath` contents to a special fallback directory. Then, the restore flows as usual. 

Reviewer: ```suggestion
You can now configure you can configure PBM to use a fallback directory during a physical restore. At the restore start, PBM copies the `dbPath` contents to a special fallback directory. Then, the restore flows as usual. 
```

If the restore is successful, PBM deletes the fallback directory. If PBM detects that the cluster is in an error state, it triggers the fallback procedure and restores the cluster to its pre-restore state. This ensures your cluster remains operational, allowing you to retry the restore from another backup or perform other maintenance tasks.

Note that this functionality requires sufficient free disk space on each `mongod` instance to store the `dbPath` contents. At the restore start, PBM checks the available disk space and logs this information.

With this feature, your cluster can remain operational even if errors occur during a physical restore. For more details about the fallback flow and configuration, see the [documentation](../features/physical.md/#physical-restores-with-a-fallback-directory).

### Improved security via the updated libraries

Percona Backup for MongoDB now uses the AWS SDK v2 for Go, which includes the latest features, security updates and improvements. This ensures the secure communication with AWS services.

> Percona Backup for MongoDB also uses `viper` and `cobra` libraries instead of a non-maintained `kingpin`. This serves to improve the user experience with PBM via the ability to create a more engaging command line interface.  

Reviewer 2: This was part of v2.9 (swap kingpin with viper/cobra), I'm not sure why we are mention it now.

Author: Well, we haven't announced it at all. Is it too internal stuff that brings no value to the users? Should I remove it at all?

Author: @radoslawszulgo please add your input. Should I keep this info or remove it?

Author: Removed this info

* [PBM-1481](https://perconadev.atlassian.net/browse/PBM-1481) - Remove the "Z" marker in the restore timestamps in pbm list and pbm status outputs (Thank you Nathan Neulinger for reporting this issue)

* [PBM-1484](https://perconadev.atlassian.net/browse/PBM-1484) - Use a dedicated Google Cloud Storage SDK Go library instead AWS SDK v1

* [PBM-1498](https://perconadev.atlassian.net/browse/PBM-1498) - Add HMAC support for GCS SDK

* [PBM-1514](https://perconadev.atlassian.net/browse/PBM-1514) - Improve resync performance for restores by retrieving only the latest record from the storage and introducing a flag to retrive the full history

* [PBM-1544](https://perconadev.atlassian.net/browse/PBM-1544) - Add a new "Status" column  to `pbm status` output to explicitly show a backup status

### Bugs Fixed

* [PBM-1238](https://perconadev.atlassian.net/browse/PBM-1238) - Fixed the error of reading the timestamp from a cluster heartbeat when parsing the physical restore status from a metadata file. If that field is missing, uninitialized, or improperly formatted in the database, the parsing fails. The issue is fixed by processing the timestamp from the metadata file if it exists

> * [PBM-1401](https://perconadev.atlassian.net/browse/PBM-1401) - Added support for the `--wait` flag for the `pbm config --file` and `pbm config --set` commands. When running these commands immediately follwoed by another operation such as start backup, using the `--wait` flag is the correct and recommended approach to ensure the storage is fully synchronized before running the next command.

Reviewer: ```suggestion
* [PBM-1401](https://perconadev.atlassian.net/browse/PBM-1401) - Added support for the `--wait` flag for the `pbm config --file` and `pbm config --set` commands. When running these commands immediately followed by another operation such as start backup, using the `--wait` flag is the correct and recommended approach to ensure the storage is fully synchronized before running the next command.
```

* [PBM-1401](https://perconadev.atlassian.net/browse/PBM-1401) - Added support for the `--wait` flag for the `pbm config --file` and `pbm config --set` commands. When running these commands immediately followed by another operation such as start backup, using the `--wait` flag is the correct and recommended approach to ensure the storage is fully synchronized before running the next command

* [PBM-1439](https://perconadev.atlassian.net/browse/PBM-1439) Fixed the issue with a backup failing due to the `maxUploadParts` setting being ignored during part size calculation for instances where the size was unknown. The issue is fixed by estimating oplog slice size using bytes/sec × duration with extra growth and a 1 GiB minimum (Thank you Rama Mekala for reporting this issue)

> * [PBM-1482](https://perconadev.atlassian.net/browse/PBM-1482) - Fixed issues with selective restores due to unclosed files causing S3 deadlocks and incomplete processing of `config.chunks` collection due to single-batch processing

Reviewer 2: this ticket is hidden (internal task only). Not sure if this relevant.

Reviewer 2: Ok, resolving, in the private communication we agreed that we don't want to open ticket because it contains customer's info, but we need to list the ticket because support team asked for that.
